packages/eas-cli/src/credentials/credentialsJson/read.ts

```diff
@@ -78,22 +86,52 @@ export async function readAndroidCredentialsAsync(projectDir: string): Promise<A
   };
 }
 
-export async function readIosCredentialsAsync(projectDir: string): Promise<iOSCredentials> {
+export async function readIosCredentialsAsync(projectDir: string): Promise<IosCredentials> {
   const credentialsJson = await readAsync(projectDir);
   if (!credentialsJson.ios) {
     throw new Error('iOS credentials are missing from credentials.json');
   }
+
+  if (!isInternalCredentialsMap(credentialsJson.ios)) {
+    return await readCredentialsForTargetAsync(projectDir, credentialsJson.ios);
+  } else {
+    const targets = Object.keys(credentialsJson.ios);
+    const targetCredentialsMap: IosTargetCredentialsMap = {};
+    for (const target of targets) {
+      targetCredentialsMap[target] = await readCredentialsForTargetAsync(
+        projectDir,
+        credentialsJson.ios[target]
+      );
+    }
+    return targetCredentialsMap;
+  }
+}
+
+function isInternalCredentialsMap(
+  ios: CredentialsJsonIosCredentials | Record<string, CredentialsJsonIosCredentials>
+): ios is Record<string, CredentialsJsonIosCredentials> {
+  return typeof ios.provisioningProfilePath !== 'string';
+}
+
+export function isCredentialsMap(ios: IosCredentials): ios is IosTargetCredentialsMap {
+  return typeof ios.provisioningProfile !== 'string';
```

Comment:
not too familiar with credentials json processing, but wouldn't you need to look at the dist cert too?

Reply:
I'd say it doesn't matter.
When you look at the definition of `IosCredentials` (`type IosCredentials = IosTargetCredentials | IosTargetCredentialsMap;`) then it could be either a map of single-target object. 
- If someone names one of the targets `provisioningProfile` it will pass the Joi validation but the value won't be a string.
- If someone doesn't provide `provisioningProfile` in a single-target project then credentials.json won't pass validation.